go/vt/vtgate/engine/filter_test.go

```diff
@@ -25,18 +25,41 @@ import (
 
 	"vitess.io/vitess/go/mysql/collations"
 	"vitess.io/vitess/go/sqltypes"
+	"vitess.io/vitess/go/vt/sqlparser"
 	"vitess.io/vitess/go/vt/vtgate/evalengine"
 )
 
+type dummyTranslator struct{}
+
+func (d dummyTranslator) ColumnLookup(col *sqlparser.ColName) (int, error) {
+	switch col.Name.String() {
+	case "left":
+		return 0, nil
+	case "right":
+		return 1, nil
+	default:
+		panic("unexpected column name")
+	}
+}
+
+func (d dummyTranslator) CollationForExpr(_ sqlparser.Expr) collations.ID {
+	return collationEnv.LookupByName("utf8mb4_bin").ID()
+}
+
+func (d dummyTranslator) DefaultCollation() collations.ID {
+	return collationEnv.LookupByName("utf8mb4_bin").ID()
```

Comment:
Any specific reason this is using the binary collation and not return whatever the default collation is? Of course it's a dummy, but was wondering if there's more behind it or not then.